include/gatekeeper_main.h

```diff
@@ -41,4 +43,16 @@ extern uint64_t cycles_per_sec;
 extern uint64_t cycles_per_ms;
 extern uint64_t picosec_per_cycle;
 
+struct log_ratelimit_state *get_log_ratelimit_state_by_lcore(void);
+
+#define RTE_LOG_RATELIMIT(l, t, ...)                        \
```

Comment:
1. This macro seems big. Could/should not it become a function? If so, `get_log_ratelimit_state_by_lcore()` doesn't need to be public.

2. The blocks now have macros for their own logging, so you need to update those macros to use the rate limit.

Reply:
Will do.